crates/types/src/message.rs

```diff
@@ -172,6 +174,8 @@ pub enum MessageKind<TYPES: NodeType> {
     Consensus(SequencingMessage<TYPES>),
     /// Messages relating to sharing data between nodes
     Data(DataMessage<TYPES>),
+    /// A (still serialized) message to be passed through to external listeners
+    External(Vec<u8>),
```

Comment:
This shouldn't break versioning since it is additive but I will have to make sure